inc/notificationtemplatetemplate.class.php

```diff
@@ -0,0 +1,361 @@
+            if ($mode === NOT_AVAILABLE) {
+               $mode = "{$data['mode']} ($mode)";
+            } else {
+               $mode = $mode['label'];
+            }
+            echo "<td>$mode</td>";
+            echo "</tr>";
+            Session::addToNavigateListItems(__CLASS__, $data['id']);
+         }
+         echo $header;
+      } else {
+         echo "<tr class='tab_bg_2'><th colspan='$colspan'>".__('No item found')."</th></tr>";
+      }
+
+      echo "</table>";
+      echo "</div>";
+   }
+
+
+   function getName($options = []) {
+      return $this->getID();
+   }
+
+
+   /**
+    * Print the form
+    *
+    * @param integer $ID      ID of the item
+    * @param array   $options array
+    *     - target for the Form
+    *     - computers_id ID of the computer for add process
+    *
+    * @return true if displayed  false if item not found or not right to display
+   **/
+   function showForm($ID, $options=array()) {
+      global $CFG_GLPI;
+
+      if (!Session::haveRight("notification", UPDATE)) {
+         return false;
+      }
+
+      $notif = new Notification();
+      if ($ID > 0) {
+         $this->check($ID, READ);
+         $notif->getFromDB($this->fields['notifications_id']);
+      } else {
+         $this->check(-1, CREATE, $options);
+         $notif->getFromDB($options['notifications_id']);
+      }
+
+      $this->showFormHeader($options);
+
+      if ($this->isNewID($ID)) {
+         echo "<input type='hidden' name='notifications_id' value='".$options['notifications_id']."'>";
+      }
+
+      echo "<tr class='tab_bg_1'>";
+      echo "<td>".__('Notification')."</td>";
+      echo "<td>".$notif->getLink()."</td>";
+      echo "<td colspan='2'>&nbsp;</td>";
+      echo "</tr>\n";
+
+      echo "<tr class='tab_bg_1'>";
+      echo "<td>" . __('Mode') . "</td>";
+      echo "<td>";
+      self::dropdownMode(['name' => 'mode', 'value' => $this->getField('mode'), 'multiple' => false]);
+      echo "</td>";
+
+      echo "<td>". NotificationTemplate::getTypeName(1)."</td>";
+      echo "<td><span id='show_templates'>";
+      NotificationTemplate::dropdownTemplates(
+         'notificationtemplates_id',
+         $notif->fields['itemtype'],
+         $this->fields['notificationtemplates_id']
+      );
+      echo "</span></td></tr>";
+
+      $this->showFormButtons($options);
+
+      return true;
+
+   }
+   /**
+    * Get notification method label
+    *
+    * @param string $mode the mode to use
+    *
+    * @return array
+   **/
+   static function getMode($mode) {
+      $tab = self::getModes();
+      if (isset($tab[$mode])) {
+         return $tab[$mode];
+      }
+      return NOT_AVAILABLE;
+   }
+
+   /**
+    * Register a new notification mode (for plugins)
+    *
+    * @param string $mode  Mode
+    * @param string $label Mode's label
+    * @param strign $from  Plugin which registers the mode
+    *
+    * @return void
+    */
+   static public function registerMode($mode, $label, $from) {
```

Comment:
It is a hook on CFG_GLPI, but not directly; the goal was to require only a few values, since only a few values are required; there is no need to write a full class here.